packages/runtime/src/interpreter/interpreter.ts

```diff
@@ -175,10 +177,15 @@ export class Interpreter {
     this.runtime = runtime;
     this.instructions = parser(program, mode, this);
 
+    let dynamicCost = 0;
     while (this.instructionIndex < this.instructions.length) {
       const instruction = this.instructions[this.instructionIndex];
       instruction.execute(this.stack);
 
+      // for teal version >= 4, cost is calculated dynamically at the time of execution
+      dynamicCost += this.lineToCost[instruction.line];
+      if (this.tealVersion >= 4) { assertMaxCost(dynamicCost, mode); }
+
```

Comment:
By doing this,  it is possible that cost is very low because if some branches of code are never executed. right ?
or this cost is addition to static cost ?

Reply:
great